### PR TITLE
Sergey Budkov 2536

### PR DESCRIPTION
**Pull request by sergeybp**

### CODE REVIEW

File: app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java [hunk 1]
```diff
@@ -10,18 +12,26 @@
...
-    int width = 0;
-    int height = 0;
-    int scale = 4;
+    int [][] field_a = null;
+    int [][] field_b = null;
+    int [][] field_c = null;
+    int width = 240;
+    int height = 320;
+    int [] field_colors = null;
+    Bitmap picture;
```
Comment (nwdeer) [inline app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java:24]: Can we do without field_colors and picture? Seems that we have all information about colors in field_a and field_b

File: app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java [hunk 2]
```diff
@@ -10,18 +12,26 @@
...
-* Created by thevery on 11/09/14.
-*/
+ * Created by thevery on 11/09/14.
+ */
 class WhirlView extends SurfaceView implements Runnable {
-    int [][] field = null;
-    int width = 0;
-    int height = 0;
-    int scale = 4;
+    int [][] field_a = null;
```
Comment (nwdeer) [inline app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java:18]: Better to use camelCase

File: app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java [hunk 3]
```diff
@@ -60,29 +71,39 @@ public void run() {
...
+                field_a [x][y] = rand.nextInt(MAX_COLOR);
             }
         }
+        field_colors = new int[width*height];
+
     }
 
     void updateField() {
-        int[][] field2 = new int[width][height];
+        int choose = 1;
```
Comment (nwdeer) [inline app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java:95]: unclear naming

File: app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java [hunk 4]
```diff
@@ -10,18 +12,26 @@
...
+    int [][] field_a = null;
+    int [][] field_b = null;
+    int [][] field_c = null;
+    int width = 240;
+    int height = 320;
+    int [] field_colors = null;
+    Bitmap picture;
+    DisplayMetrics displaymetrics = getResources().getDisplayMetrics();
+    int m1 = displaymetrics.widthPixels;
+    int m2 = displaymetrics.heightPixels;
```
Comment (thevery) [inline app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java:27]: bad naming

File: app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java [hunk 5]
```diff
@@ -10,18 +12,26 @@
...
 class WhirlView extends SurfaceView implements Runnable {
-    int [][] field = null;
-    int width = 0;
-    int height = 0;
-    int scale = 4;
+    int [][] field_a = null;
+    int [][] field_b = null;
+    int [][] field_c = null;
+    int width = 240;
+    int height = 320;
```
Comment (thevery) [inline app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java:22]: `public static final WIDTH = 320;`

File: app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java [hunk 6]
```diff
@@ -90,24 +111,27 @@ void updateField() {
...
     @Override
     public void onDraw(Canvas canvas) {
-        for (int x=0; x<width; x++) {
-            for (int y=0; y<height; y++) {
-                Paint paint = new Paint();
-                paint.setColor(palette[field[x][y]]);
-                canvas.drawRect(x*scale, y*scale, (x+1)*scale, (y+1)*scale, paint);
-            }
-        }
+        canvas.scale(width_to_draw,height_to_draw);
```
Comment (thevery) [inline app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java:131]: reformat your code with ctrl/cmd+alt+L

**Comment by thevery** [comment]
ok = 8
~27 FPS

**Pull request closed without merging**